include/swift/Driver/Action.h

```diff
@@ -168,6 +168,26 @@ class CompileJobAction : public JobAction {
   static bool classof(const Action *A) {
     return A->getKind() == Action::Kind::CompileJob;
   }
+
+  /// Return a _single_ TY_Swift InputAction, if one exists;
+  /// if 0 or >1 such inputs exist, return nullptr.
+  const InputAction *findSingleSwiftInput() const {
+    auto Inputs = getInputs();
+    const InputAction *IA = nullptr;
+    for (auto const *I : Inputs) {
+      if (auto const *S = dyn_cast<InputAction>(I)) {
```

Comment:
Nit: The nesting here is pretty deep.  This can be refactored into a straight-line loop.

Reply:
I'm not entirely sure how to fix it, without  a `compactMap` operation, but I have taken a stab at rewriting it for clarity.